### ISSUE TITLE
`no-navigation-without-base` reports links starting with `#`

### ISSUE THREAD

**Issue by marekdedic** [sveltejs/eslint-plugin-svelte#1104]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.

### What version of ESLint are you using?

9

### What version of `eslint-plugin-svelte` are you using?

3.0.2

### What did you do?

```svelte
<a href="#section">Click me</a>
```

### What did you expect to happen?

No error

### What actually happened?

`no-navigation-without-base` reports an error

### Link to **GitHub Repo** with Minimal Reproducible Example

N/A, will make a PR instead
